source/ranges.tex

```diff
@@ -1071,19 +1132,25 @@
 
 \pnum
 The name \tcode{ranges::cdata} denotes a customization point
-object\iref{customization.point.object}. The expression
-\tcode{ranges::\brk{}cdata(E)} for a subexpression \tcode{E} of type \tcode{T}
-is expression-equivalent to:
+object\iref{customization.point.object}.
+Given a subexpression \tcode{E} with type \tcode{T},
+let \tcode{t} be an lvalue that denotes the reified object for \tcode{E}.
+Then:
 \begin{itemize}
-\item \tcode{ranges::data(static_cast<const T\&>(E))} if \tcode{E} is an lvalue.
-
-\item Otherwise, \tcode{ranges::data(static_cast<const T\&\&>(E))}.
+\item
+If \tcode{E} is an rvalue and
+\tcode{enable_borrowed_range<remove_cv_t<T>>} is \tcode{false},
+\tcode{ranges::cdata(E)} is ill-formed.
+\item
+Otherwise,
+\tcode{ranges::cdata(E)} is expression-equivalent to
+\tcode{\exposid{as-const-pointer}(ranges::data(\exposid{possi\-bly-const-range}(t)))}.
```

Comment:
The definition of _`as-const-pointer`_ at the start of this subclause is missing.

Reply:
fixed